### ISSUE TITLE
Input text tokenizer

### ISSUE THREAD

**Issue by grzegorzgoldapl** [learn-and-code/Manbot#4]
Bot needs to recognize commands and arguments to them. Commands will most likely have a fixed structure with some specific number of arguments.

This task is to write a converter from an input string coming from user into a list of strings comprising arguments. It doesn't suffice to split words with spaces, because we might need a quoted string with spaces as an one argument, so for example:

For input string:

> !project new "My Project"

We don't want to interpret it as

> [!project, new, "My, Project"]

But as

> [!project, new, My Project]

# Specification

A token is a string meeting of these requirements:

> SQ_TOKEN: starts with a ', ends with ', all other characters allowed in between
> DQ_TOKEN: starts with a ", end with ", all other characters allowed in between
> SIMPLE_TOKEN: can contain letters, digits, special characters excep whitespaces, ", '

There can be whitespaces between tokens and they are to be ignored, so if we have a line like:

> token1 Token2 token3

Tokens are:

> [token1, Token2, token3]

all of kind SIMPLE_TOKEN. Spaces between them are ignored

and in a line like

> token1 "quoted token2" 'single "quoted" token3'

Tokens are:

> [token1, quoted token2, single "quoted" token3]

of kinds SIMPLE_TOKEN, DQ_TOKEN and SQ_TOKEN. Again, spaces between tokens are ignored.

# Class specification

We want an interface:

```Java
interface IInputTextTokenizer {
    bool isCommandCandidate(String input);
    List<String> tokenize(String input);
}
```
and a concrete class implementing this interface and above specification:

```Java
class SimpleInputTextTokenizer : IInputTextTokenizer {
    public bool isCommandCandidate(String input) { /* ... */ }
    public List<String> tokenize(String input) { /* ... */ }
}
```

Method isCommandCandidate is used to test if the input string has the valid format of command. In our case, we recognize the command by checking if the first token is of type SIMPLE_TOKEN, and if this token starts with a ! character. So this function should split the whole string into tokens, just to check if the first one meets the criteria for command.

Please implement this in a branch **input-text-tokenizer** created from the **testing** branch.

**Comment by TheProBro** [comment]
I didnt understand my work

**Comment by kevin51jiang** [comment]
@grzegorzgoldapl Can you assign me to this task as well?

**Comment by grzegorzgoldapl** [comment]
@kevin51jiang in a separate branch, where you come up with your implementation? Or do you want to work with @DylanReedy on the same piece of code?

**Comment by kevin51jiang** [comment]
@grzegorzgoldapl I'm gonna work with @DylanReedy . I think we were both assigned the same task in the Discord.

**Comment by grzegorzgoldapl** [comment]
Oh, ok. Please communicate with @DylanReedy then, but yeah, you are assigned.

**Comment by grzegorzgoldapl** [comment]
@kevin51jiang No, wait, there is confusion here. It's not the task that was mentioned on the discord by @gavt13 This issue is for different task, assigned to @DylanReedy earlier.
